### PR TITLE
Transports integration

### LINKED ISSUE #1229: Extract TransportConfiguration to a separate assembly
Currently `TransportConfiguration` is defined within the ServiceControl assembly. This means that the transport packages need a reference on the full ServiceControl project aand it also means that ServiceControl can't have a reference to the transport packages. This makes it hard to provide a F5 solution.

Moving `TransportConfiguration` to a separate assembly will allow ServiceControl to have a reference on the transport projects and enable F5 of ServiceControl

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes #1229 

Introduces a dedicated `ServiceControl.Transports` assembly that SC references and transports implement

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/servicecontrol-maintainers any good ideas on how to resolve the dependency on `Settings` in the custom check as well as the dependency to `SettingsReader`? I was thinking of some clever duck typing over endpoint configuration but I'm not yet sure how to address this properly

File: src/ServiceControl/ServiceControl.csproj [hunk 1]
```diff
@@ -50,4 +50,7 @@
     <EmbeddedResource Include="Infrastructure\RavenDB\RavenLicense.xml" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\ServiceControl.Transports\ServiceControl.Transports.csproj" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (bording) [reply src/ServiceControl/ServiceControl.csproj]: More generally, the layout is:
```
Property group with general project properties (with TargetFramework as the first entry)

Project references

Assembly references

Package references

ItemGroups related to files in the project

Custom Targets
```

[the remark this answers is not in the document]
Comment (danielmarbach) [reply src/ServiceControl/ServiceControl.csproj]: 👍 will align

**Comment by danielmarbach** [comment]
One approach could be that we map the important settings to the transport settings as spiked here. There is still some weirdness going on with bootstrapper, setup bootstrapper and that the responsibilities of assigning settings are spread around but I don't think this makes it much worse since the problem was already there

**Comment by bording** [comment]
@danielmarbach With the changes here, is there any more package reference removal that can be done for the main ServiceControl project?

**Comment by danielmarbach** [comment]
Well in theory we could get rid of NSB raw and NSB reference in SC since SC
refernces the transports abstraction. But I think we should first agree
whether this is a path to pursue. Alternatively would could leave status
quo of develop and deploy MSMQ customization as part of debug build

Brandon Ording <notifications@github.com> schrieb am Sa., 14. Juli 2018,
20:52:

**Pull request merged**